File: Proyecto-Prog_Avanzada_Frontend-master/src/app/modelo/publicacion-get-dto.ts
import { ComentarioGetDTO } from "./comentario-get-dto";
import { DetalleCompraGetDTO } from "./detalle-compra-get-dto";
import { ProductoGetDTO } from "./producto-get-dto";
import { UsuarioGetDTO } from "./usuario-get-dto";

export class PublicacionGetDTO {
    codigo: number = 0;
    codigoCuenta: number = 0;
    codigoProducto: number = 0;
    fechaLimite!: Date;
    descuento: number = 0;
    estado: string = "";
    comentarios: ComentarioGetDTO[];
    detalleCompras: DetalleCompraGetDTO[];
    producto: ProductoGetDTO;

    constructor(codigo: number, vendedor: number, descuento: number, fechaLimite: Date, estado: string, comentarios: ComentarioGetDTO[], codigoProducto: number, detalleCompras: DetalleCompraGetDTO[], producto: ProductoGetDTO) {
        this.codigo = codigo;
        this.codigoCuenta = vendedor;
        this.descuento = descuento;
        this.fechaLimite = fechaLimite;
        this.estado = estado;
        this.comentarios = comentarios;
        this.codigoProducto = codigoProducto;
        this.detalleCompras = detalleCompras;
        this.producto = producto;
    }
}

File: Proyecto-Prog_Avanzada-railway/src/main/java/co/edu/uniquindio/proyecto/proyectotienda/dto/PublicacionGetDTO.java
package co.edu.uniquindio.proyecto.proyectotienda.dto;

import co.edu.uniquindio.proyecto.proyectotienda.modelo.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.List;

@AllArgsConstructor
@Getter
@Setter
public class PublicacionGetDTO {
    private int codigo;

    private int codigoCuenta;

    private int codigoProducto;

    private LocalDateTime fechaLimite;

    private int descuento;

    private EstadoPublicacion estado;

    private List<ComentarioGetDTO> comentarios;

    private List<DetalleCompraGetDTO> detalleCompras;

    private ProductoGetDTO producto;
}
